### ISSUE TITLE
Crashes after `TextArea::set_content()` with an exit code 101

### ISSUE THREAD

**Issue by edubenetskiy** [gyscos/Cursive#164]
## Description

After invoking the `TextArea::set_content()` method and then bringing back the focus to the text area using the keyboard arrow keys, the program may crash with an exit code 101.

This issue is similar to #147.

## Code to reproduce

```rust
extern crate cursive;

use cursive::Cursive;
use cursive::traits::Identifiable;
use cursive::views::{Dialog, TextArea};

fn main() {
    let mut siv = Cursive::new();

    let text_area = TextArea::new().with_id("input");
    siv.add_layer(Dialog::around(text_area)
        .button("Clear", |s| {
            s.call_on_id("input", |i: &mut TextArea| i.set_content(""));
        })
        .button("Quit", |s| s.quit()));

    siv.run();
}
```
## Steps to reproduce

1. Launch the program with `$ cargo run`
2. **Type something** in the text area
    - If you leave the text area empty, the following actions will not lead to a failure.
3. Press the **“Clear”** button
    - You can focus the button with either your mouse or keyboard – it does not affect subsequent failure.
4. **Navigate back** to the text area **using the arrow keys** or `Tab`, not with the mouse
    - If you click the TextArea with the mouse at this step, it focuses normally and does not fail.
5. The application quits the *ncurses* mode and crashes with an exit code 101.

## Configuration

```
$ lsb_release -ds
"Arch Linux"

$ uname -srm
Linux 4.12.3-1-user-ns-enabled x86_64

$ rustc --version
rustc 1.22.1 (05e2e1c41 2017-11-22)
```

### Cargo.toml

```toml
# ...
[dependencies]
cursive = { git = "https://github.com/gyscos/Cursive" }
```

**Comment by edubenetskiy** [comment]
I have tried with both *ncurses* and Termion backends and it still crashes.

**Comment by gyscos** [comment]
Thank you very much for the report!
Indeed, `TextArea::set_content` was pretty careless - it should be fixed in fdf4a0db4b3d1250bbab3363df33722f663e2b8d.

**Comment by edubenetskiy** [comment]
Allow me to express my gratitude for such a prompt response. I have found it an enjoyable experience to use Cursive, thank you for maintaining the library.
